litmus/director/upgrade-replica-pod-pending/test.yml

```diff
@@ -0,0 +1,167 @@
+            method: GET
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+          register: volume_health
+          until: "volume_health.json.data[0].data.pods[0].volumes[0].replica[0].state=='Healthy'"
+          retries: 20
+          delay: 2
+        
+        ## Get application volume health status for replica-2
+        - name: Get application volume health status for replica-2
+          uri:
+            url: '{{ director_url }}/v3/groups/{{ group_id }}/clusters/{{ cluster_id }}/mayaapplications'
+            method: GET
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+          register: volume_health
+          until: "volume_health.json.data[0].data.pods[0].volumes[0].replica[1].state=='Healthy'"
+          retries: 20
+          delay: 2
+
+        ## Get application volume health status for replica-3
+        - name: Get application volume health status for replica-3
+          uri:
+            url: '{{ director_url }}/v3/groups/{{ group_id }}/clusters/{{ cluster_id }}/mayaapplications'
+            method: GET
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+          register: volume_health
+          until: "volume_health.json.data[0].data.pods[0].volumes[0].replica[2].state=='Healthy'"
+          retries: 20
+          delay: 2
+        
+        ## Get replicas of application volumes
+        - name: Get replicas of application volumes
+          shell: >
+            kubectl get pods --all-namespaces -l {{ pod_label }} -o jsonpath='{.items[0].metadata.name}'
+          register: pod_name
+
+        ## Fetch namespace of replica pod
+        - name: Fetch namespace of replica pod
+          shell: >
+            kubectl get pods --all-namespaces -l {{ pod_label }} -o jsonpath='{.items[0].metadata.namespace}'
+          register: pod_namespace
+        
+        ##  Get the node_name of corresponding job
+        - name: Get the node_name of corresponding job
+          shell: >
+            kubectl get pod {{ pod_name.stdout }} -n {{ pod_namespace.stdout }} -o jsonpath={.spec.nodeName}
+          register: node_name
+        
+        ## Taint the node in which the pod is scheduled
+        - name: Taint the node in which pod is scheduled
+          shell: >
+            kubectl taint nodes {{ node_name.stdout }} l=b:NoSchedule
+        
+        ## Delete the replica pod of volume
+        - name: Delete the replica pod of volume
+          shell: > 
+            kubectl delete pod {{ pod_name.stdout }} -n {{ pod_namespace.stdout }}
```

Comment:
How does this ensure that replica pod will never run?
Are we assuming that these pod have anti affinity & 3 is the replica count of application & 3 is total node count.